packages/ui/src/working-groups/modals/CreateOpening/CreateOpening.tsx

```diff
@@ -0,0 +1,169 @@
+import { machineStateConverter } from '@/council/modals/AnnounceCandidacy/helpers'
+import { useMyMemberships } from '@/memberships/hooks/useMyMemberships'
+import { StyledStepperBody } from '@/proposals/modals/AddNewProposal'
+import { GroupIdToGroupParam } from '@/working-groups/constants'
+
+import { SuccessModal, CreateOpeningSteps as Steps, ImportOpening } from './components'
+import { createOpeningMachine, CreateOpeningMachineState, getTxParams } from './machine'
+import { OpeningConditions, CreateOpeningForm, CreateOpeningModalCall, OpeningSchema, defaultValues } from './types'
+
+export const CreateOpeningModal = () => {
+  const [showImport, setShowImport] = useState<boolean>(false)
+
+  const { api, connectionState } = useApi()
+  const { active: activeMember } = useMyMemberships()
+  const [state, send, service] = useMachine(createOpeningMachine)
+  const { hideModal, modalData } = useModal<CreateOpeningModalCall>()
+
+  const { group } = modalData
+  const context = { hiringTarget: 1, minStake: new BN(50000), group } as OpeningConditions
+  const path = useMemo(() => machineStateConverter(state.value), [state.value])
+  const resolver = useYupValidationResolver<CreateOpeningForm>(OpeningSchema, path)
+  const form = useForm<CreateOpeningForm>({ resolver, mode: 'onChange', defaultValues, context })
+  useEffect(() => {
+    form.trigger(machineStateConverter(state.value) as keyof CreateOpeningForm)
+  }, [machineStateConverter(state.value)])
+
+  const createOpeningTx = useMemo(() => {
+    if (api && group) {
+      const { ...specifics } = form.getValues() as CreateOpeningForm
+      const { description, stakePolicy, rewardPerBlock } = getTxParams(group, specifics)
+      return api.tx[group].addOpening(description, 'Regular', stakePolicy, String(rewardPerBlock))
+    }
+  }, [connectionState, activeMember?.id, form.formState.isValidating, group])
+
+  const { feeInfo } = useTransactionFee(activeMember?.controllerAccount, () => createOpeningTx, [
+    connectionState,
+    createOpeningTx,
+  ])
+  const setExportJsonValue = () => {
+    const { ...specifics } = form.getValues() as CreateOpeningForm
+    const exportValue = {
+      applicationDetails: specifics.durationAndProcess.details,
+      title: specifics.workingGroupAndDescription.title,
+      shortDescription: specifics.workingGroupAndDescription.shortDescription,
+      description: specifics.workingGroupAndDescription.description,
+      applicationFormQuestions: specifics.applicationForm?.questions?.map((question) => {
+        return { question: question.questionField }
+      }),
+      stakingPolicy: {
+        amount: specifics.stakingPolicyAndReward.stakingAmount / Math.pow(10, JOY_DECIMAL_PLACES),
+        unstakingPeriod: specifics.stakingPolicyAndReward.leavingUnstakingPeriod,
+      },
+      rewardPerBlock: specifics.stakingPolicyAndReward.rewardPerBlock / Math.pow(10, JOY_DECIMAL_PLACES),
+    }
+    return exportValue
```

Comment:
Also since this is in a `useMemo` lets return the string value:
```suggestion
    return JSON.stringify(exportValue)
```